core/src/main/java/org/apache/gravitino/cache/CacheUtils.java

```diff
@@ -0,0 +1,280 @@
+    String metalakeName = getMetalakeName(cache, modelPO.getMetalakeId());
+    String catalogName = getCatalogName(cache, modelPO.getCatalogId());
+    String schemaName = getSchemaName(cache, modelPO.getSchemaId());
+
+    return POConverters.fromModelPO(
+        modelPO, NamespaceUtil.ofModel(metalakeName, catalogName, schemaName));
+  }
+
+  /**
+   * Returns the topic entity from the topic PO.
+   *
+   * @param cache The cache to retrieve the topic entity from.
+   * @param topicPO The topic PO to convert to an entity.
+   * @return The {@link TopicEntity} instance.
+   */
+  public static TopicEntity getTopicEntityFromPO(MetaCache cache, TopicPO topicPO) {
+    String metalakeName = getMetalakeName(cache, topicPO.getMetalakeId());
+    String catalogName = getCatalogName(cache, topicPO.getCatalogId());
+    String schemaName = getSchemaName(cache, topicPO.getSchemaId());
+
+    return POConverters.fromTopicPO(
+        topicPO, NamespaceUtil.ofTopic(metalakeName, catalogName, schemaName));
+  }
+
+  /**
+   * Returns the fileset entity from the fileset PO.
+   *
+   * @param cache The cache to retrieve the fileset entity from.
+   * @param filesetPO The fileset PO to convert to an entity.
+   * @return The {@link FilesetEntity} instance.
+   */
+  public static FilesetEntity getFilesetEntityFromPO(MetaCache cache, FilesetPO filesetPO) {
+    String metalakeName = getMetalakeName(cache, filesetPO.getMetalakeId());
+    String catalogName = getCatalogName(cache, filesetPO.getCatalogId());
+    String schemaName = getSchemaName(cache, filesetPO.getSchemaId());
+
+    return POConverters.fromFilesetPO(
+        filesetPO, NamespaceUtil.ofFileset(metalakeName, catalogName, schemaName));
+  }
+
+  /**
+   * Returns the tag entity from the tag PO.
+   *
+   * @param cache The cache to retrieve the tag entity from.
+   * @param tagPO The tag PO to convert to an entity.
+   * @return The {@link TagEntity} instance.
+   */
+  public static TagEntity getTagEntityFromPO(MetaCache cache, TagPO tagPO) {
+    String metalakeName = getMetalakeName(cache, tagPO.getMetalakeId());
+
+    return POConverters.fromTagPO(tagPO, NamespaceUtil.ofTag(metalakeName));
+  }
+
+  /**
+   * Returns the name of the metalake based on its id.
+   *
+   * @param cache The cache to retrieve the metalake name from.
+   * @param metalakeId The id of the metalake.
+   * @return if the metalake is in the cache, returns the name of the metalake. Otherwise, returns
+   *     the name of the metalake from the database.
+   */
+  public static String getMetalakeName(MetaCache cache, Long metalakeId) {
+    if (cache.containsById(metalakeId, Entity.EntityType.METALAKE)) {
+      return getIdentFromEntity(cache.getOrLoadMetadataById(metalakeId, Entity.EntityType.METALAKE))
+          .name();
+    } else {
+      return MetalakeMetaService.getInstance().getMetalakePOById(metalakeId).getMetalakeName();
+    }
```

Comment:
Maybe we can directly call `cache.getOrLoadMetadataById(metalakeId, Entity.EntityType.METALAKE)` to replace this code?